python/lsst/jointcal/dataIds.py

```diff
@@ -102,7 +102,7 @@ def makeDataRefList(self, namespace):
                     self._addDataRef(namespace, ref.dataId, tract)
 
         # Ensure all components of a visit are kept together by putting them all in the same set of tracts
-        for visit, tractSet in visitTract.items():
+        for visit, tractSet in sorted(visitTract.items()):
```

Comment:
Good catch! tractSet is a `set()`, so would have the same problem. I don't have any test data that crosses sets, so I never encountered this.

That said, maintaining order here is only to keep tests passing between py2 and py3 while I sort out *why* there's order dependence. I've added a note to that effect above the nested loops.